### PR TITLE
Fix block parsing 

### PR DESCRIPTION
**Pull request by maraoz**

### CODE REVIEW

**Comment by martindale** [comment]
Reviewing – fix those tests, tho'.

File: test/script/script.js [hunk 1]
```diff
@@ -755,4 +753,5 @@ describe('Script', function() {
     });
   });
 
+
```
Comment (martindale) [inline test/script/script.js]: nit: extra space

File: test/script/interpreter.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ Script.fromBitcoindString = function(str) {
       var tstr = token.slice(1, token.length - 1);
       var cbuf = new Buffer(tstr);
       tbuf = Script().add(cbuf).toBuffer();
+      //bw.writeUInt8(tstr.length);
```
Comment (martindale) [inline test/script/interpreter.js]: nit: commented line

File: lib/transaction/input/input.js [hunk 3]
```diff
@@ -105,14 +114,19 @@ Input.prototype.toBufferWriter = function(writer) {
 };
 
 Input.prototype.setScript = function(script) {
+  this._script = null;
   if (script instanceof Script) {
     this._script = script;
     this._scriptBuffer = script.toBuffer();
+  } else if (JSUtil.isHexa(script)) {
+    // hex string script
+    this._scriptBuffer = new Buffer(script, 'hex');
```
Comment (braydonf) [inline lib/transaction/input/input.js]: Should `this._script` should be set here?

Comment (maraoz) [inline lib/transaction/input/input.js]: No, it's only set where it's already there. In this case, it will be lazy loaded if requested

**Comment by braydonf** [comment]
LGTM

**Pull request merged**